core/frontend/src/main/scala/io/udash/bindings/modifiers/ValueModifier.scala

```diff
@@ -1,35 +1,45 @@
 package io.udash.bindings.modifiers
 
+import com.avsystem.commons.SharedExtensions._
 import io.udash.bindings._
 import io.udash.properties._
 import io.udash.properties.single.ReadableProperty
 import io.udash.utils.Registration
 import org.scalajs.dom
 import org.scalajs.dom._
 
-import scalatags.generic._
+import scala.scalajs.js
 
-private[bindings] trait ValueModifier[T] extends Modifier[dom.Element] with Bindings {
-  def property: ReadableProperty[T]
-  def builder: (T => Seq[dom.Element])
-  def checkNull: Boolean
-  def listen(callback: T => Unit): Registration
+private[bindings]
+trait ValueModifier[T] extends Binding {
+  import Bindings._
+
+  protected def property: ReadableProperty[T]
+  protected def builder: ((T, Binding => Binding) => Seq[dom.Element])
+  protected def checkNull: Boolean
+  protected def listen(callback: T => Unit): Registration
+
+  protected def replace(root: Element)(oldElements: Seq[Element], newElements: Seq[Element]): Unit =
+    root.replaceChildren(oldElements, newElements)
 
   override def applyTo(t: dom.Element): Unit = {
-    var elements: Seq[Element] = null
+    var elements: Seq[Element] = Seq.empty
```

Comment:
I'd rather see this as an explicitly mutable collection but with an immutable reference.

Reply:
It is always replaced with new sequence of elements. I think it's easier to understand than `clear` and `++=` combination.